src/tools/quemap/lightmap.c

```diff
@@ -1015,50 +1042,42 @@ void FinalizeLighting(int32_t face_num) {
 		// apply brightness, saturation and contrast
 		ColorFilter(lightmap, lightmap, brightness, saturation, contrast);
 
-		// write the lightmap sample data as bytes
-		for (int32_t j = 0; j < 3; j++) {
-			*dest++ = (byte) Clamp(lightmap[j] * 255, 0, 255);
-		}
+		if (legacy) { // write out good old RGB lightmap samples, converted to bytes
+			for (int32_t j = 0; j < 3; j++) {
+				*dest++ = (byte) Clamp(floor(lightmap[j] * 255.0 + 0.5), 0, 255);
+			}
+		} else { // write out HDR lightmaps and deluxemaps
+
+			ColorEncodeRGBM(lightmap, hdr_lightmap);
+
+			for (int32_t j = 0; j < 4; j++) {
+				*dest++ = (byte) Clamp(floor(hdr_lightmap[j] * 255.0 + 0.5), 0, 255);
+			}
 
-		if (!legacy) { // also write the directional data
 			vec3_t direction, deluxemap;
+			vec4_t hdr_deluxemap;
 
 			// start with the raw direction data
 			VectorCopy((fl->directions + i * 3), direction);
 
 			// if the sample was lit, it will have a directional vector in world space
 			if (!VectorCompare(direction, vec3_origin)) {
-				vec3_t normal;
-
-				if (tex->flags & SURF_PHONG) { // interpolated normal
-					SampleNormal(f, fl->origins + i * 3, normal);
-				} else {
-					if (f->side) {
-						VectorNegate(bsp_file.planes[f->plane_num].normal, normal);
-					} else {
-						VectorCopy(bsp_file.planes[f->plane_num].normal, normal);
-					}
-				}
```

Comment:
Ok, this is what I was asking about over Discord. So you basically tackle this when the sample is gathered, rather than when it is finalized. Works for me 👍